### DISCUSSION TITLE
dynamodbstreams: `ApproximateCreationDateTime` returns `datetime.datetime` Instead of timestamp

### DISCUSSION THREAD

**Discussion by RajasGujarathi** [spulec/moto#8042] [Q&A] [answered]
#### **moto** version - `5.0.13`
---

For DynamoDB Streams, `dynamodbstreams`, I'm encountering an issue with the `dynamodbstreams.get_records(ShardIterator=...)` method. 

Specifically, the response contains a field `ApproximateCreationDateTime` with a value of `datetime.datetime(...)`, whereas as per [AWS docs](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_streams_StreamRecord.html) it is expected to be in ISO 8601 timestamp format.

I get the following response
```
{'Records': [{'eventID': '7d712df1b71b475da61a2f7aa17c7020', 'eventName': 'INSERT', 'eventVersion': '1.0', 'eventSource': 'aws:dynamodb', 'awsRegion': 'us-east-1', 'dynamodb': {'ApproximateCreationDateTime': datetime.datetime(2024, 8, 27, 15, 16, 17, 606284), .....
```

Can someone help confirm whether this is a bug or if I am doing something wrong?

**Comment by bblommers** [comment, marked as answer]
Hi @RajasGujarathi! The HTTP API will indeed send a ISO-8601 string, but boto3 will automatically convert that into a `datetime` object.
See the boto3-specific API here: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodbstreams/client/get_records.html

For background information: Moto actually returns the date in ISO format as well, because we only intercept/mock the actual HTTP response. `boto3` still converts the Moto response into a `datetime` object, like it would do with an actual AWS response.

https://github.com/getmoto/moto/blob/035dfacb1d7e76d1e8f9ed7a071704241f73aa92/moto/dynamodbstreams/models.py#L92
